scripts/install.sh

```diff
@@ -1,159 +1,298 @@
-  case "$name" in
-    h)
+  case "$OPT" in
+    h | help)
       help
       exit 0
       ;;
-    m) INSTALL_NETWORK_MANAGER="true"
+    l | list-versions)
+      get_versions
+      exit 0
       ;;
-    n) DISABLE_NETWORKING="true"
+    v | version)
+      needs_arg
+      VERSION=$(echo "$OPTARG" | sed 's/^v//')  # drop leading 'v's
       ;;
-    q) QUIET="true"
+    a | arch) needs_arg; ARCH=$OPTARG
+      ;;
+    m | install-nm)
+      INSTALL_NETWORK_MANAGER="$(echo ${OPTARG:-'yes'} | tr '[:upper:]' '[:lower:]')"
+      case "$INSTALL_NETWORK_MANAGER" in
+        yes)
+          ;;
+        no)
+          ;;
+        ask)
+          ;;
+        * )
+          die "Valid options for -m, --install-nm are: 'yes', 'no', and 'ask'"
+          ;;
+      esac
+      ;;
+    n | no-networking) DISABLE_NETWORKING="true"
+      ;;
+    q | quiet) QUIET="true"
+      ;;
+    \?)  # Handle invalid short options
+      die "Error: Invalid option -$OPTARG" \
+          "See './install.sh -h' for more information."
+      ;;
+    * )  # Handle invalid long options
+      die "Error: Invalid option --$OPT" \
+          "See './install.sh -h' for more information."
       ;;
-    \?)
-      echo "Error: Invalid option -- '$OPTARG'"
-      echo "Try './install.sh -h' for more information."
-      exit 1
   esac
 done
 
 shift $(($OPTIND -1))
 
 if [ "$(id -u)" != "0" ]; then
-   echo "This script must be run as root" 1>&2
-   exit 1
+   die "This script must be run as root"
+fi
+
+if [[ -z "$ARCH" ]]; then
+  debug "Arch was not specified. Inferring..."
+  ARCH=$(uname -m)
+  debug "Arch was inferred to be $ARCH"
 fi
 
-ARCH=$(uname -m)
 ARCH_NAME=""
 if [ "$ARCH" = "aarch64" ]; then
   ARCH_NAME="linuxarm64"
 elif [ "$ARCH" = "armv7l" ]; then
-  echo "ARM32 is not supported by PhotonVision. Exiting."
-  exit 1
+  die "ARM32 is not supported by PhotonVision. Exiting."
 elif [ "$ARCH" = "x86_64" ]; then
   ARCH_NAME="linuxx64"
 else
-  if [ "$#" -ne 1 ]; then
-      echo "Can't determine current arch; please provide it (one of):"
-      echo ""
-      echo "- linuxarm64 (64-bit Linux ARM)"
-      echo "- linuxx64   (64-bit Linux)"
-      exit 1
-  else
-    echo "Can't detect arch (got $ARCH) -- using user-provided $1"
-    ARCH_NAME=$1
-  fi
+  die "Unsupported or unknown architecture: '$ARCH'." \
+  "Please specify your architecture using: ./install.sh -a <arch> " \
+  "Run './install.sh -h' for more information."
 fi
 
-echo "This is the installation script for PhotonVision."
-echo "Installing for platform $ARCH_NAME"
+debug "This is the installation script for PhotonVision."
+debug "Installing for platform $ARCH"
 
 DISTRO=$(lsb_release -is)
-if [[ "$DISTRO" = "Ubuntu" && "$INSTALL_NETWORK_MANAGER" != "true" && -z "$QUIET" && -z "$DISABLE_NETWORKING" ]]; then
-  echo ""
-  echo "Photonvision uses NetworkManager to control networking on your device."
+
+if [[ "$DISTRO" != "Ubuntu" || -n "$DISABLE_NETWORKING" || -n "$QUIET" ]] ; then
```

Comment:
I think that the logic here will override `--install-nm=yes` if the `--quiet` flag is also passed.

Reply:
Ah yes you're right. Let me fix that.